src/app/clusters/commodity-price-server/commodity-price-server.cpp

```diff
@@ -128,24 +161,326 @@ void Instance::InvokeCommand(HandlerContext & handlerContext)
 void Instance::HandleGetDetailedPriceRequest(HandlerContext & ctx,
                                              const Commands::GetDetailedPriceRequest::DecodableType & commandData)
 {
-    //    bool validArgs = false;
+    DataModel::Nullable<Structs::CommodityPriceStruct::Type> priceStruct;
+
+    chip::BitMask<CommodityPriceDetailBitmap> details = commandData.details;
+    if (!details.HasOnly(
+            BitMask<CommodityPriceDetailBitmap>(CommodityPriceDetailBitmap::kDescription, CommodityPriceDetailBitmap::kComponents)))
+    {
+        ChipLogError(Zcl, "Invalid details bitmap recvd in GetDetailedPriceRequest");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        return;
+    }
 
-    //    map16 details                       = commandData.details;
+    Commands::GetDetailedPriceResponse::Type response;
 
-    // TODO finish off
+    CHIP_ERROR err = GetDetailedPriceRequest(details, priceStruct);
```

Comment:
`priceStruct` should be declared here.

Though your could also just do `GetDetailedPriceRequest(details, response.currentPrice);` directly, right?